### PR TITLE
Fix return value of DataDeskRequireTagParameter(s). Fix array declara…

### PR DESCRIPTION
**Pull request by thePHTest**
…tion causing graph output to spit out a bunch of extra newlines.

### CODE REVIEW

**Comment by thePHTest** [comment]
My apologies for not syncing to master first. I'll make sure to do so next time. Fortunately it looks like it can auto-merge just fine :)

**Review by ryanfleury — CHANGES_REQUESTED** [review]
This isn't quite correct; `DataDeskRequireTagParameter` should allow 0 for a `result` parameter.

File: source/data_desk.h [hunk 1]
```diff
@@ -849,11 +849,11 @@ DataDeskRequireTagParameter(DataDeskNode *root, char *tag, int index, DataDeskNo
     if(tag_node)
     {
         DataDeskNode *tag_param = DataDeskGetTagParameter(tag_node, index);
-        if(result)
+        if(tag_param)
```
Comment (thePHTest) [inline source/data_desk.h:875]: So are you thinking:
`if(tag_param)
{
  if(result)
  {
    *result = tag_param;
  }
  found = 1;
}`

And this way you can query for the required parameter without having a result allocated.

Comment (ryanfleury) [reply source/data_desk.h:875]: Yep, that's the idea. So you don't need to pass in a pointer if you are just querying the existence of something (which would just be annoying boilerplate).

**Pull request merged**